docs/reference/models.md

```diff
@@ -0,0 +1,183 @@
+        Course ||--o{ ProductCourseRelation : "is included in"
+        
+        Product ||--o{ ProductCourseRelation : ""
+        
+        ProductCourseRelation ||..|| OrderCourseRelation : "copied at order creation"
+```
+
+### Focus
+
+#### `course` and `target_courses` fields of `Product` and `Order` models
+
+The `course` field represents the course in which the product is sold or the order is owned.
+On the other hand, the `target_courses` field represents the courses contained in the product/order.
+
+#### The `ProductCourseRelation` and `OrderCourseRelation` models
+
+First the `ProductCourseRelation` model is a join table between  a `Product` and a `Course`.
+
+It has extra attributes :
+- `position`: To order courses in the product
+- `is_graded`: To take in account the grade of the course for the certificate delivery
+- `course_runs`: To restrict the course to specific course runs
+
+Then the `OrderCourseRelation` model is a join table between an `Order` and a `Course`.
+This relation is created at the creation of an `Order` by copying the `CourseProductRelation`
+of the related `Product`. In this way, a `Product` can be updated without impacting previously
+created `Order`.
+
+#### The signal `on_add_course_runs_to_product_course_relation`
+
+It corresponds to a `m2m_changed` signal that is fired before a course run is added to a
+product course relation. It is in charge to prevent to add a course run to a product
+course relation if those two resources do not rely on the same course.
+
+
+## Relation between `Order` / `ProformaInvoice` and `Transaction` models
+
+```mermaid
+erDiagram
+    %% Models
+    Order {
+        Integer id PK
+        User owner FK
+        Course course FK
+    }
+
+    ProformaInvoice {
+        Integer id PK
+        Order order FK
+        ProformaInvoice parent FK
+    }
+    
+    Transaction {
+        Integer id PK
+        ProformaInvoice proforma_invoice FK
+    }
+    
+    %% Relations
+    Order ||--|| only_one_proforma_invoice_without_parent_per_order : ""
+    only_one_proforma_invoice_without_parent_per_order ||--o{ ProformaInvoice : "has"
+    
+    ProformaInvoice ||--o{ ProformaInvoice : "is child of"
+    ProformaInvoice ||--|| Transaction : "reconcils"
+```
+
+### Focus
+
+#### Nesting of `ProformaInvoice`
+
+The `ProformaInvoice` model has a field `parent` to be able to link proforma invoice between them.
+An `Order` should have only one proforma invoice without parent which is the main invoice. Then if
+order owner pays in installments, a new proforma invoice is created for each of them and linked to
+main proforma invoice. Also if we need to refund the client a proforma invoice
+with a negative amount (a credit note) is created.
+
+#### Constraints on `ProformaInvoice` model
+
+The `ProformaInvoice` model has two database constraint :
+
+- A `UniqueConstraint` "only_one_proforma_invoice_without_parent_per_order" to prevent an order to 
+have several main proforma invoices.
```

Comment:
```suggestion
- A `UniqueConstraint` "only_one_proforma_invoice_without_parent_per_order" to prevent an order 
from having several main proforma invoices.
```